tempo_embeddings/embeddings/model.py

```diff
@@ -298,10 +308,10 @@ def mean_pooling(model_output, attention_mask):
         )
 
     @torch.no_grad()
-    def _passage_embeddings(
-        self, passages: list[Passage], store_tokenizations: bool
+    def embed_passages(
+        self, passages: list[Passage], store_tokenizations: bool, skip_batching: bool = False
```

Comment:
Based on the suggestion above (move `_batches()` to the `Corpus` class), this could replace the `skip_batching` parameter with `batch_size`. If `batch_size` is set to `None`, this implies to skip batching.
I suppose the `passages` parameter then would have to become `corpus`.

```suggestion
    def embed_passages(
        self, corpus: Corpus, store_tokenizations: bool, batch_size: Optional[int]
```